tests/fixture/tmpnet/node_process.go

```diff
@@ -331,3 +341,60 @@ func (p *NodeProcess) writeMonitoringConfigFile(tmpnetDir string, name string, c
 
 	return nil
 }
+
+// watchLogFileForFatal waits for the specified file path to exist and then checks each of
+// its lines for the string 'FATAL' until such a line is observed or the provided context
+// is canceled. If line containing 'FATAL' is encountered, it will be provided as an error
+// to the provided cancelWithCause function.
+func watchLogFileForFatal(ctx context.Context, cancelWithCause context.CancelCauseFunc, w io.Writer, path string) {
+	waitInterval := 100 * time.Millisecond
+	// Wait for the file to exist
+	fileExists := false
+	for !fileExists {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			if _, err := os.Stat(path); os.IsNotExist(err) {
+				// File does not exist yet - wait and try again
+				time.Sleep(waitInterval)
+			} else {
+				fileExists = true
+			}
+		}
+	}
+
+	// Open the file
+	file, err := os.Open(path)
+	if err != nil {
+		_, _ = fmt.Fprintf(w, "failed to open %s: %v", path, err)
+		return
+	}
+	defer file.Close()
+
+	// Scan for lines in the file containing 'FATAL'
+	reader := bufio.NewReader(file)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			// Read a line from the file
+			line, err := reader.ReadString('\n')
+			if err != nil {
+				if err.Error() == "EOF" {
+					// If end of file is reached, wait and try again
+					time.Sleep(waitInterval)
+					continue
+				} else {
+					_, _ = fmt.Fprintf(w, "error reading %s: %v\n", path, err)
+					return
+				}
+			}
+			if strings.Contains(line, "FATAL") {
```

Comment:
we should probably consider adding a flag that would fail the test by default unless specified

Reply:
Because if we have a fatal in the logs and the test passes silently, we might have a bug we're overlooking.